src/tasks/05_countProducts.js

```diff
@@ -15,3 +15,8 @@
  */
 
 // <-- Тут реализация countProducts
+export function countProducts(array) {
+    let smallArray =  array.filter((item) => item.discount !== null);
+    let result = { count: Object.keys(array).length, withDiscount: Object.keys(smallArray).length };
```

Comment:
Нет необходимости выполнять Object.keys, у массива уже есть свойство .length
```suggestion
    let result = { count: array.length, withDiscount: smallArray.length };
```